### PR TITLE
Build user-agent

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Fix https://github.com/pulumi/pulumi-google-native/issues/73

Local result: `Pulumi/3.4.1-0.20210610083513-3e170f6f8001 (GPN:Pulumi; https://www.pulumi.com) pulumi-google-native/0.4.1-alpha.1624003951+75cfc5b4.dirty`

We emailed Google folks to verify whether GPN should be "Pulumi" or something else, may need to adjust as the response arrives.

### CODE REVIEW

File: provider/pkg/provider/provider.go [hunk 1]
```diff
@@ -648,6 +656,31 @@ func (p *googleCloudProvider) getConfig(configName, envName string) string {
...
+
+func getPulumiVersion() string {
+	if bi, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range bi.Deps {
+			if strings.HasPrefix(dep.Path, "github.com/pulumi/pulumi/pkg") {
+				return strings.TrimPrefix(dep.Version, "v")
+			}
+		}
+	}
+	// We should never get here but let's not panic and return something sensible if we do.
```
Comment (viveklak) [inline provider/pkg/provider/provider.go:680]: Log something if we run into this?

**Review by viveklak — APPROVED** [review]

**Review by infin8x — APPROVED** [review]
LGTM, pending any verification/responses from Google on the format of the UA string

File: provider/pkg/provider/provider.go [hunk 2]
```diff
@@ -648,6 +656,31 @@ func (p *googleCloudProvider) getConfig(configName, envName string) string {
...
+	customPartnerID := p.getConfig("partnerNumber", "GOOGLE_PARTNER_NUMBER")
+	if customPartnerID != "" {
+		return customPartnerID
+	} else {
+		disablePartnerID := p.getConfig("disablePartnerNumber", "GOOGLE_DISABLE_PARTNER_NUMBER")
+		if disablePartnerID == "true" {
+			return ""
+		}
+	}
+	return "Pulumi"
```
Comment (lblackstone) [inline provider/pkg/provider/provider.go:669]: It's a little odd to return a string for a "partner number". Is this correct?

**Review by lblackstone — APPROVED** [review]

File: provider/pkg/provider/provider.go (hunk 2 above)
Comment (mikhailshilkov) [reply provider/pkg/provider/provider.go:669]: It is odd. They call it "google partner number" but just confirmed it should be "Pulumi" 🤷

Comment (mikhailshilkov) [reply provider/pkg/provider/provider.go:669]: Okay, maybe it's actually "google partner name". I'll change the name in the PR.

File: provider/pkg/provider/provider.go (hunk 1 above)
Comment (mikhailshilkov) [reply provider/pkg/provider/provider.go:680]: Done

**Pull request merged**